roles/app/templates/nginx/sites-available/readthedocs-main.conf

```diff
@@ -0,0 +1,214 @@
+        add_header X-Served Nginx;
+        add_header X-Deity docs;
+    }
+
+    # document serving
+    location ~* ^/(?P<publisher>[^/]+)/(?P<project>[^/]+)/(?P<document>[^/]+)/ {
+        # first submatch redirect to the default language / version
+        location ~* ^/(?P<publisher>[^/]+)/(?P<project>[^/]+)/(?P<document>[^/]+)/$ {
+            add_header X-Redirct-From Nginx;
+            add_header X-Deity docs;
+            rewrite ^ https://test.docs.local/$publisher/$project/$document/{{ rtd_default_language }}/{{ rtd_default_version }}/;
+        }
+
+        # second submatch serves any project file directly from the local disk. path group is needed to build the
+        # try_files argument because we can't use $uri as we need to remove the prefix up to project
+        location ~* ^/(?P<publisher>[^/]+)/(?P<project>[^/]+)/(?P<document>[^/]+)/(?P<lang>\w\w)/(?P<version>[^/]+)/(?P<path>.*)$ {
+            root /home/docs/docs.italia.it/public_web_root/;
+            try_files /$document/$lang/$version/$path /$document/$lang/$version/index.html =404;
+        }
+    }
+
+    # publisher / projects home page
+    location ~* ^/(?P<publisher>[^/]+)/(?P<project>[^/]+)/ {
+        try_files $uri $uri/ @fallback;
+        error_page 403 404 500 = @fallback;
+        add_header X-Subdomain-TryFiles True;
+        add_header X-Served Nginx;
+        add_header X-Deity docs;
+    }
+
+    location ~* ^/(?P<publisher>[^/]+)/ {
+        try_files $uri $uri/ @fallback;
+        error_page 403 404 500 = @fallback;
+        add_header X-Subdomain-TryFiles True;
+        add_header X-Served Nginx;
+        add_header X-Deity docs;
+    }
+
+    location @fallback {
+        proxy_pass http://{{ rtd_app_server }}:8001;
+        proxy_buffering off;
+        proxy_set_header Host $http_host;
+        proxy_set_header X-Real-IP $remote_addr;
+        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+        proxy_set_header X-Forwarded-Host $host;
+        add_header X-Fallback True;
+        add_header X-Served Django;
+        add_header X-Deity docs;
+    }
+
+    # Sendfile support for serving from Python
+    location /user_builds/ {
+        internal;
+        root {{ rtd_root }};
+        add_header X-Served Nginx-Sendfile;
+        add_header X-Deity docs;
+    }
+    location /public_web_root {
+        internal;
+        root {{ rtd_root }};
+        add_header X-Served Nginx-Sendfile;
+        add_header X-Deity docs;
+    }
+    # Sendfile support for serving media artifacts in Python
+    location /prod_artifacts/ {
+        internal;
+        root   {{ rtd_root }};
+        add_header X-Served Nginx-Sendfile;
+        add_header X-Deity docs;
+    }
+
+    # Let's encrypt
+    location ^~ /.well-known/acme-challenge/ {
+        default_type "text/plain";
+        root /var/www/letsencrypt;
+    }
+
+    ssl_certificate /etc/letsencrypt/live/docs-italia-staging.teamdigitale.it/fullchain.pem; # managed by Certbot
```

Comment:
at least the domain should be a variable no? OTOH If it's managed by certbot we can probably remove it.

Reply:
Agree. Fixed